soul-plugin/soul-plugin-motan/src/main/java/org/dromara/soul/plugin/motan/proxy/MotanProxyService.java

```diff
@@ -0,0 +1,96 @@
+import org.dromara.soul.common.utils.GsonUtils;
+import org.dromara.soul.plugin.motan.cache.ApplicationConfigCache;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Motan proxy service.
+ *
+ * @author tydhot
+ */
+public class MotanProxyService {
+
+    /**
+     * Generic invoker object.
+     *
+     * @param body the body
+     * @param metaData the meta data
+     * @param exchange the exchange
+     * @return the object
+     * @throws SoulException the soul exception
+     */
+    @SneakyThrows
+    public Mono<Object> genericInvoker(final String body, final MetaData metaData, final ServerWebExchange exchange) throws SoulException {
+        RefererConfig<CommonHandler> reference = ApplicationConfigCache.getInstance().get(metaData.getPath());
+        if (Objects.isNull(reference) || StringUtils.isEmpty(reference.getServiceInterface())) {
+            ApplicationConfigCache.getInstance().invalidate(metaData.getPath());
+            reference = ApplicationConfigCache.getInstance().initRef(metaData);
+        }
+        CommonHandler commonHandler = reference.getRef();
+        ApplicationConfigCache.MotanParamInfo motanParamInfo = ApplicationConfigCache.PARAM_MAP.get(metaData.getMethodName());
+        Object[] params;
+        if (motanParamInfo == null) {
+            params = new Object[0];
+        } else {
+            int num = motanParamInfo.getParamTypes().length;
+            params = new Object[num];
+            for (int i = 0; i < num; i++) {
+                Map<String, Object> bodyMap = GsonUtils.getInstance().convertToMap(body);
+                params[i] = bodyMap.get(motanParamInfo.getParamNames()[i]).toString();
+            }
+        }
+        ResponseFuture responseFuture = (ResponseFuture) commonHandler.asyncCall(metaData.getMethodName(),
+                params, Object.class);
+        CompletableFuture<Object> future = CompletableFuture.supplyAsync(() -> responseFuture.getValue());
+        return Mono.fromFuture(future.thenApply(ret -> {
+            if (Objects.isNull(ret)) {
+                ret = Constants.MOTAN_RPC_RESULT_EMPTY;
+            }
+            exchange.getAttributes().put(Constants.MOTAN_RPC_RESULT, ret);
+            exchange.getAttributes().put(Constants.CLIENT_RESPONSE_RESULT_TYPE, ResultEnum.SUCCESS.getName());
+            return ret;
+        })).onErrorMap(SoulException::new);
+    }
+
+//    private GenericMessage buildGenericMessage(String name, Map<String, Object> map) {
+//        GenericMessage message = new GenericMessage();
+//        message.setName(name);
+//        for (Map.Entry<String, Object> e : map.entrySet()) {
+//            int nameIndex = CommonSerializer.getHash(e.getKey());
+//            message.putFields(nameIndex,e.getValue());
+//        }
+//        return message;
+//    }
```

Comment:
Should we remove this?

Reply:
These codes will be used to support complex object in Motan later. Refer to https://github.com/weibocom/motan/issues/892#issuecomment-599349769

Reply:
OK Let's keep this for now.